factom_keys/id.py

```diff
@@ -0,0 +1,197 @@
+import ed25519
+from bitcoin import base58
+from hashlib import sha256
+
+PREFIX_LENGTH = 3
+CHECKSUM_LENGTH = 4
+BODY_LENGTH = 35
+TOTAL_LENGTH = 39
+
+
+def generate_id_pairs():
+    """
+    Generate a set of Factom ID keypairs, sk1-sk4 and id1-id4.
+
+    :return: a list of tuples containing a random id key pair: (IDPrivateKey, IDPublicKey)
+    """
+    pairs = []
+
+    for i in range(0, 4):
+        signer, verifier = ed25519.create_keypair()
+        pair = (IDPrivateKey(signer.to_seed()), IDPublicKey(verifier.to_bytes()))
+        pairs.append(pair)
+
+    return pairs
+
+
+def _to_base58_string(prefixed_key: bytes):
+    """
+    Convert prefixed_key bytes into sk1/id strings with a checksum.
+
+    :param prefixed_key: the sk1 private key or id public key prefixed with the appropriate bytes
+    :return: a sk1 private key string or id public key string
+    """
+    prefix = prefixed_key[:PREFIX_LENGTH]
+    assert prefix in IDPublicKey.PREFIXES or prefix in IDPrivateKey.PREFIXES, "Invalid key prefix."
+    temp_hash = sha256(prefixed_key[:BODY_LENGTH]).digest()
+    checksum = sha256(temp_hash).digest()[:CHECKSUM_LENGTH]
+    return base58.encode(prefixed_key + checksum)
+
+
+class BadKeyStringError(Exception):
+    pass
+
+
+class IDPrivateKey:
```

Comment:
I would prefer the name be more indicative of the intended use (i.e. that these are specifically for ANO Server Identities).

Something like `ServerIdentityPrivateKey` and `ServerIdentityPublicKey`. That way we do not have any confusion when DIDs are widespread in factom

Reply:
I like `ServerIdentity...` for naming. I'll use that.